dashboard/modules/job/common.py

```diff
@@ -97,30 +95,34 @@ class JobInfoStorageClient:
     JOB_DATA_KEY_PREFIX = f"{ray_constants.RAY_INTERNAL_NAMESPACE_PREFIX}job_info_"
     JOB_DATA_KEY = f"{JOB_DATA_KEY_PREFIX}{{job_id}}"
 
-    def __init__(self):
+    def __init__(self, gcs_aio_client: GcsAioClient):
+        self._gcs_aio_client = gcs_aio_client
         assert _internal_kv_initialized()
 
-    def put_info(self, job_id: str, data: JobInfo):
-        _internal_kv_put(
-            self.JOB_DATA_KEY.format(job_id=job_id),
+    async def put_info(self, job_id: str, data: JobInfo):
+        await self._gcs_aio_client.internal_kv_put(
+            self.JOB_DATA_KEY.format(job_id=job_id).encode(),
```

Comment:
Btw, do you know what happens if you don't specify `encode `? Does it just fail?